### ISSUE TITLE
Add parallel downloading of audio and video

### ISSUE THREAD

**Issue by cheater** [rg3/youtube-dl#15569]
### Make sure you are using the *latest* version: run `youtube-dl --version` and ensure your version is *2018.02.08*. If it's not, read [this FAQ entry](https://github.com/rg3/youtube-dl/blob/master/README.md#how-do-i-update-youtube-dl) and update. Issues with outdated version will be rejected.
- [x] I've **verified** and **I assure** that I'm running youtube-dl **2018.02.08**

### Before submitting an *issue* make sure you have:
- [x] At least skimmed through the [README](https://github.com/rg3/youtube-dl/blob/master/README.md), **most notably** the [FAQ](https://github.com/rg3/youtube-dl#faq) and [BUGS](https://github.com/rg3/youtube-dl#bugs) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?type=Issues) the bugtracker for similar issues including closed ones
- [x] Checked that provided video/audio/playlist URLs (if any) are alive and playable in a browser

### What is the purpose of your *issue*?
- [ ] Bug report (encountered problems with youtube-dl)
- [ ] Site support request (request for adding support for a new site)
- [x] Feature request (request for a new functionality)
- [ ] Question
- [ ] Other

### Description of your *issue*, suggested solution and other information

I would like youtube-dl to download audio and video in parallel, if it decides to download both separately. Currently I have to wait for the video to download, be done, then for the audio to download, and only then does the muxing begin. The download would be much faster if the audio were downloaded while the video was being downloaded.

**Comment by cheater** [comment]
@dstftw what is this a duplicate of? I have searched the bug tracker before posting and have found nothing.

**Comment by cheater** [comment]
Just to clarify: this is not about "having parallel downloads". It's about downloading a *single* video, where youtube-dl ends up downloading two files, one for audio, and one for video. I would like this to happen in parallel.

**Comment by dstftw** [comment]
#6319.
